### PR TITLE
Add documentation for Reference Container extension

### PR DESCRIPTION
**Pull request by tom-anders**
corresponding patch: https://reviews.llvm.org/D137894

relevant issue: https://github.com/clangd/clangd/issues/177

### CODE REVIEW

File: extensions.md [hunk 1]
```diff
@@ -254,6 +254,20 @@ tracking allocations directly).
...
+{:.v16}
+
+When finding references to a symbol, it's often useful to know the name of the
+function or class in which the reference occurs. 
+
+**New property of `Location` object**: `containerName : string?`:
+ - Name of the function or class in which the reference occurs. Might be empty,
+   e.g. for global scope.
+
+**New client capability**: `textDocument.references.contaiiner`:
```
Comment (HighCommander4) [inline extensions.md:267]: typo (`contaiiner`)

**Review by HighCommander4 — APPROVED** [review]

**Comment by tom-anders** [comment]
Thanks! I don't think I have the necessary permissions, so you'd have to merge this for me

**Comment by HighCommander4** [comment]
I don't have commit access to this repository either, let's ask @kadircet

File: extensions.md [hunk 2]
```diff
@@ -254,6 +254,20 @@ tracking allocations directly).
...
  - Signals that the server supports `$/memoryUsage` requests.
 
+## Reference Container
+{:.v16}
+
+When finding references to a symbol, it's often useful to know the name of the
+function or class in which the reference occurs. 
+
+**New property of `Location` object**: `containerName : string?`:
+ - Name of the function or class in which the reference occurs. Might be empty,
```
Comment (kadircet) [inline extensions.md:264]: could you clarify the difference between empty vs none/missing (if any)?

Comment (tom-anders) [reply extensions.md]: Done: our current implementation actually never sends none/missing here if the `textDocument.references.container` capability is set, so I changed the type from `string?` to `string`.

Comment (tom-anders) [reply extensions.md]: Sorry, you're absolutely right! I for some reason thought that the index would just return an invalid/empty symbol in this case, so we would fill the containerName with an empty string still, but that's not the case.

Hope the wording is clear now.

File: extensions.md [hunk 3]
```diff
@@ -254,6 +254,21 @@ tracking allocations directly).
...
 
+## Reference Container
+{:.v16}
+
+When finding references to a symbol, it's often useful to know the name of the
+function or class in which the reference occurs. 
+
+**New property of `Location` object**: `containerName : string?`:
+ - Name of the function or class in which the reference occurs. Might be null,
+   e.g. if the containing symbol is indexed. Can also be an empty string, e.g.
```
Comment (kadircet) [inline extensions.md:265]:
```suggestion
   e.g. if the containing symbol is not indexed. Can also be an empty string, e.g.
```

**Review by kadircet — APPROVED** [review]

**Comment by kadircet** [comment]
thanks!

**Pull request merged**